File: src/main/scala/com/vc/easy/L443.scala
package com.vc.easy

object L443 {
  def compress(arr: Array[Char]): Int = {
    var i = 0
    var j = 0
    var count = 0
    val n = arr.length
    while(i < n) {
      val ch = arr(i)
      count = 0
      while(i < n && ch == arr(i)) {
        count += 1
        i += 1
      }
      arr(j) = ch
      j += 1
      if(count > 1) {
        count.toString.foreach(ch => {
          arr(j) = ch
          j += 1
        })
      }
    }
    j
  }
}

File: src/main/java/com/vc/easy/L443.java
package com.vc.easy;

class L443 {
    public int compress(char[] chars) {
        if(chars == null || chars.length == 0) return 0;

        int index = 0, count = 0, insertIndex = 0;
        while(index < chars.length) {
            char ch = chars[index];
            count = 0;
            while(index < chars.length && chars[index] == ch) {
                index++;
                count++;
            }
            chars[insertIndex++] = ch;
            if(count > 1) {
                String countString = String.valueOf(count);
                for(int i = 0; i < countString.length(); i++)
                    chars[insertIndex++] = countString.charAt(i);
            }
        }
        return insertIndex;
    }
}
